### PR TITLE
ITS-3342 Use API1694 Data to make request to NPS SendLetter.

### PR DESCRIPTION
**Pull request by rambabu-posa**

### CODE REVIEW

**Review by ajchilts — COMMENTED** [review]
This looks like some great work and providing it tests OK in an integrated environment we should be able to go with this approach. 

We have added some new services and repositories and they should have associated unit / integration tests. 

I understand this will take a little longer but I think it's important to add tests to give confidence in the new repo's etc. 

I am out tomorrow and as such will not restrict this with 'request changes' however, feel we could add some confidence through some additional integration / unit tests.

**Comment by rambabu-posa** [comment]
Thanks @ajchilts  we are still working on unit tests. We just want to review it quickly when you're available today.

File: it/repositories/EncryptedIndividualDetailsRepositoryISpec.scala [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+      "when there is a record for this nino" - {
+
+        "must get the record" in {
+
+          repository.insertOrReplaceIndividualDetailsData(individualDetailsData).futureValue
+          val result = repository.findIndividualDetailsDataByNino(individualDetailsData.getNino).futureValue
+          result.value.copy(lastUpdated = Instant.EPOCH) mustEqual individualDetailsData.copy(lastUpdated = Instant.EPOCH)
+        }
+      }
```
Comment (Shihabur-Atos) [inline it/repositories/EncryptedIndividualDetailsRepositoryISpec.scala:69]: Missing a test scenario where the record isn't found

File: test/services/IndividualDetailsServiceSpec.scala [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    val nino = "AB123456C"
+    val individualDetailsDataCache = IndividualDetailsDataCache("12345", Some(IndividualDetailsData("John", "Doe", "1980-01-01", "AB12CD", "AB123456C")))
+
+    when(mockRepo.findIndividualDetailsDataByNino(any())(any()))
+      .thenReturn(Future.successful(Some(individualDetailsDataCache)))
+
+    val result = service.getIndividualDetailsData(nino)
+
+    result.futureValue shouldBe Some(individualDetailsDataCache)
+  }
```
Comment (Shihabur-Atos) [inline test/services/IndividualDetailsServiceSpec.scala:113]: Missing a test scenario where the record isn't found

**Pull request merged**